tests/vector/vecswiz.cpp

```diff
@@ -0,0 +1,52 @@
+/* RUN: %{execute}%s
+
+   Test some vec<> simple swizzles behaviour
+*/
+#define SYCL_SIMPLE_SWIZZLES
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <boost/test/minimal.hpp>
+
+using namespace cl::sycl;
+
+int test_main(int argc, char *argv[]) {
+	vec<int, 4> v = {1, 2, 3, 4};
+
+	/* test an xyzw index */
+	BOOST_CHECK(v.x() == 1);
+	BOOST_CHECK(v.y() == 2);
+	BOOST_CHECK(v.z() == 3);
+	BOOST_CHECK(v.w() == 4);
+
+	/* test a rgba index */
+	BOOST_CHECK(v.r() == 1);
+	BOOST_CHECK(v.g() == 2);
+	BOOST_CHECK(v.b() == 3);
+	BOOST_CHECK(v.a() == 4);
+
+	/* test a xyzw swizzle */
+	auto v2 = v.zyxx();
+	BOOST_CHECK(v2.x() == 3);
+	BOOST_CHECK(v2.y() == 2);
```

Comment:
Missing
```
	BOOST_CHECK(v2.z() == 1);
	BOOST_CHECK(v2.w() == 1);
```

Reply:
done for next pass.

Reply:
Breaking news: according to https://gitlab.khronos.org/sycl/Specification/issues/160 it is not correct to use `auto` to store the result of a swizzle...

So there are a lot of `auto` that should be explicited in this file.